### ISSUE TITLE
"Basic" authentication doesn't return WWW-Authenticate header on failure

### ISSUE THREAD

**Issue by geogeim** [cdimascio/express-openapi-validator#471]
I registered a basic security schema with my own validateSecurity.handlers for basic auth. When the user enters the URL in the browser he doesn't get the login prompt because the authentication doesn't return the "WWW-Authenticate" header in the response and just fails with "Authorization header required"

That means i have to work around it by adding this in the global error handler

```
      if(error.path === '/users/export' && error.status === 401) {
        res.set('WWW-Authenticate', 'Basic realm="ugh"');
      }
```
Which kinda defeats the purpose of the security handler in the first place :(

**Comment by cdimascio** [comment]
@geogeim good point. the challenge here is where to find the realm. Potentially, we can provide the response object to the security handler. open to ideas

**Comment by ahilke** [comment]
@cdimascio  What about just using `WWW-Authenticate: Basic`? According to the [specification](https://tools.ietf.org/html/rfc7235#appendix-C), this should be valid.

[MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate) says: 

> realm=<realm\>
    A description of the protected area. If no realm is specified, clients often display a formatted hostname instead.

I verified this behaviour in Chromium and Firefox working like intended, i.e. showing a login popup. I tested it with this [demo project](https://github.com/ahilke/nestjs-with-express-openapi-validator/tree/express-openapi-validator-471) (although it lacks documentation).

I never came across a use case for using realms, never mind multiple realms, so I cannot say what the implications would be. But it would allow to cover the basic and common use case without needing to add any attribute to the OpenAPI spec.

If you agree, I could take a stab at this sometime.

**Comment by cdimascio** [comment]
That will be great. Please do. It makes for a good launching point.
